### PR TITLE
mgr/dashboard_v2: Added status route to cluster

### PR DESCRIPTION
**Pull request by sebastian-philipp**
* `RESTController`: added `@detail_route`
* Fixed `RequestHelper.assertJsonBody`

Signed-off-by: Sebastian Wagner <sebastian.wagner@suse.com>

### CODE REVIEW

File: src/pybind/mgr/dashboard_v2/tests/test_tools.py [hunk 1]
```diff
@@ -30,6 +30,14 @@ def delete(self, key):
     def bulk_delete(self):
         FooResource.elems = []
 
+    def set(self, data, key):
+        FooResource.elems[int(key)] = data
+        return dict(key=key, **data)
+
+    @detail_route(methods=['get'])
```
Comment (rjfd) [inline src/pybind/mgr/dashboard_v2/tests/test_tools.py:37]: why not use the cherrypy annotations like:

```
@cherrypy.expose
def detail(self, key):
    # ....
```

Comment (sebastian-philipp) [reply src/pybind/mgr/dashboard_v2/tests/test_tools.py:37]: I don't know! Is it possible to expose `/api/cluster/<fsid>/status`, but not  `/api/cluster/status` with `@cherrypy.expose` ?

Comment (rjfd) [reply src/pybind/mgr/dashboard_v2/tests/test_tools.py:37]: I don't know either

Comment (sebastian-philipp) [reply src/pybind/mgr/dashboard_v2/tests/test_tools.py:37]: Nope doesn't work: `"The path '/foo/1/detail2' was not found."`

File: src/pybind/mgr/dashboard_v2/tests/helper.py [hunk 2]
```diff
@@ -32,8 +33,15 @@ def _delete(self, url, data=None):
     def _put(self, url, data=None):
         self._request(url, 'PUT', data)
 
-    def assertJsonBody(self, data):
-        self.assertBody(json.dumps(data))
+    def assertJsonBody(self, data, msg=None):
```
Comment (rjfd) [inline src/pybind/mgr/dashboard_v2/tests/helper.py:36]: why do you need the `msg` parameter, it's not being used anywhere AFAICT

Comment (sebastian-philipp) [reply src/pybind/mgr/dashboard_v2/tests/helper.py:36]: Copy & paste of the existing code. I didn't want to mess with the logic in detail. Instead, leave it as it is.

Comment (rjfd) [reply src/pybind/mgr/dashboard_v2/tests/helper.py:36]: C&P from where?

File: src/pybind/mgr/dashboard_v2/tools.py [hunk 3]
```diff
@@ -71,6 +71,7 @@ def load_controllers(mgrmodule):
 
 
 def _json_error_page(status, message, traceback, version):
+    cherrypy.serving.response.headers['Content-Type'] = 'application/json'
```
Comment (p-se) [inline src/pybind/mgr/dashboard_v2/tools.py]: Will be superseded by #50. I also don't know why we need the `serving` in there (as it works well without it).

**Review by p-se — APPROVED** [review]

File: src/pybind/mgr/dashboard_v2/tools.py (hunk 3 above)
Comment (votdev) [reply src/pybind/mgr/dashboard_v2/tools.py]: cherrypy.request and cherrypy.response are redirected to cherrypy.serving, so it does not matter if you use it or not. But i think we should prefer NOT to use cherrypy.serving.xxx.

**Review by s0nea — APPROVED** [review]

File: src/pybind/mgr/dashboard_v2/tests/helper.py (hunk 2 above)
Comment (sebastian-philipp) [reply src/pybind/mgr/dashboard_v2/tests/helper.py:36]: `cheroot.test.webtest.WebCase#assertBody`

**Review by rjfd — APPROVED** [review]
lgtm

**Pull request merged**